### PR TITLE
Binned maximum likelihood estimation

### LINKED ISSUE #91: Add MLE parameter estimation

### LINKED ISSUE #94: Remove VTR control parameter from DEoptim
VTR incorrect parameter:

> `VTR` :    the value to be reached. The optimization process will stop if either the maximum number of iterations itermax is reached or the best parameter vector bestmem has found a value fn(bestmem) <= VTR. Default to -Inf.

Correct parameter should be

> `reltol` relative convergence tolerance. The algorithm stops if it is unable to reduce the value by a factor of reltol * (abs(val) + reltol) after steptol steps. Defaults to sqrt(.Machine$double.eps), typically about 1e-8.

with `steptol` set to `< itermax`

> `steptol` see reltol. Defaults to itermax.

### PR DESCRIPTION
**Pull request by stefaneng**
# Description
Add support for MLE on binned data. Note: Does _not_ refactor code to work on Histogram class and does MLE estimation like bin widths are size 1.

### Closes #91 and closes #94

# Checklist

- [x] This PR does NOT contain [PHI](https://ohrpp.research.ucla.edu/hipaa/) or germline genetic data.  A repo may need to be deleted if such data is uploaded. Disclosing PHI is a [major problem](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m).
- [x] This PR does NOT contain molecular files, compressed files, output files such as images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other non-plain-text files.  To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example.
- [x] I have read the [code review guidelines](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List).
- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://confluence.mednet.ucla.edu/pages/viewpage.action?spaceKey=BOUTROSLAB&title=GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.
- [x] The name of the branch is meaningful and well formatted following the [standards](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
- [ ] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

File: R/metric.R [hunk 1]
```diff
@@ -26,28 +26,69 @@ histogram.jaccard <- function(x, y) {
...
+  } else {
+    return((1 / (b - a))^N)
+  }
+}
+
+# x numeric
+# Assumes bins are (0, 1], (1, 2], ... (N - 1, N]
+#' @exportS3Method uniform.mle numeric
+uniform.mle.numeric <- function(x, a, b, inclusive = TRUE, log = TRUE) {
+  N <- sum(x)
```
Comment (helen-zhu) [inline R/metric.R:79]: this is probably redundant

File: R/metric.R [hunk 2]
```diff
@@ -26,28 +26,69 @@ histogram.jaccard <- function(x, y) {
...
 histogram.chisq <- function(x, y) {
   sum((x - y)^2 / (x + y))
 }
+
+uniform.mle <- function(x, a, b, inclusive = TRUE, log = TRUE) {
+  UseMethod('uniform.mle')
+}
+
+# Internal, generalizes functionality between uniform.mle methods
+uniform.mle.helper <- function(x, x.start, x.end, a, b, inclusive = TRUE, log = TRUE) {
```
Comment (helen-zhu) [inline R/metric.R:60]: Just out of curiosity, why did we use "uniform.mle" instead of "uniform_mle" for these functions?

File: R/mle.R [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#'   upper = c(max(bin.x$interval_end), 10),
+#'   control = list(
+#'     trace = TRUE,
+#'     itermax = 500
+#'     )
+#'   )
+#' print(dist_optim$optim$bestmem)
+bin_log_likelihood <- function(theta, cdf, counts, bin_lower, bin_upper, ...) {
+  res <- unlist(lapply(seq_along(counts), function(i) {
+    bin_prob <- cdf(bin_upper[i], theta[1], theta[2], ...) - cdf(bin_lower[i], theta[1], theta[2], ...)
```
Comment (helen-zhu) [inline R/mle.R:45]: Can we have more than 2 parameters (i.e. theta[1] and theta[2]) here or is that not applicable to any of the distributions?

**Review by helen-zhu — APPROVED** [review]
Generally looks terrific! I was just curious about a few things but nothing major :)

File: R/metric.R (hunk 2 above)
Comment (stefaneng) [reply R/metric.R:60]: Eventually we will switch `_` to `.` for all function for BL standard.

File: R/mle.R (hunk 3 above)
Comment (stefaneng) [reply R/mle.R:45]: We don't have any 3 parameter distributions but we could allow it to take arbitrary number of parameters.

**Pull request merged**